src/app_service/service/settings/service.nim

```diff
@@ -312,13 +329,20 @@ QtObject:
   proc saveSendStatusUpdates*(self: Service, value: bool): bool =
     var newStatus = StatusType.Online
     if not value:
-      newStatus = StatusType.Offline
+      # This looks unintuitive, but `StatusType.Offline` is interpreted
+      # as `StatusUpdate.UNKNOWN_STATUS_TYPE` in status-go, which causes an error.
+      # Hence `Invisible` (= 4 [which is `INACTIVE` in status-go]) is considerd offline.
+      newStatus = StatusType.Invisible
 
     try:
-      let r = status_update.setUserStatus(int(newStatus))
-      if(self.saveSetting(KEY_SEND_STATUS_UPDATES, value)):
-        self.settings.sendStatusUpdates = value
-        return true
+      # The new user status needs to always be broadcast, so we need to update
+      # the settings accordingly and might turn it off afterwards (if user has 
+      # set status to "offline"
+      if (self.saveSetting(KEY_SEND_STATUS_UPDATES, true)):
```

Comment:
@PascalPrecht even after reading your comment above I am not sure that I understood the point here. :)
Not sure that I get it why do we need to set online status to `true` firstly? Do we broadcast status change via `saveSetting` or `setUserStatus` call?

Wondering may we have something like this?
```
if(self.saveSetting(KEY_SEND_STATUS_UPDATES, value)):
    discard status_update.setUserStatus(int(newStatus)) ## maybe check a response here, not sure what we have in that response
    self.settings.sendStatusUpdates = value
    return true
```

Reply:
Alright, so the issue is that when we `setUserStatus()` we also broadcast the user status. This is needed to inform other users about your online status, but also to sync your online status across paired devices.

There's a little catch though: `setUserStatus()` only broadcasts when the corresponding flag allows for it.
When the user status was set to offline, the next time you set it to online, it won't be broadcast, because after setting it offline first, the flag for broadcast is turned off as well.

As a result, we always have to broadcast at least once no matter what your online status is going to be, **then** update the broadcast flag (which is what's happening here)

Reply:
Ok, thanks, though that's a bit weird flow... maybe we should improve that in future, or at least add new param "forceUpdate" (defaulted to false), but anyhow whatever user/app(in case of inactivity) sets as status should be broadcasted always.

Reply:
I believe part of that issue is that the boadcast flag is the same flag that also periodically broadcasts the status (every 5 min I believe).

You don't want/need to broadcast your online status every 5 min when you're "offline".
So yea, a force broadcast option would be good!